AppCenter/AppCenter/Internals/Util/MSOrderedDictionary.m

```diff
@@ -0,0 +1,69 @@
+#import "MSOrderedDictionaryPrivate.h"
+
+@implementation MSOrderedDictionary
+
+- (instancetype)init {
+  if ((self = [super init])) {
+    _dictionary = [NSMutableDictionary new];
+    _order = [NSMutableArray new];
+  }
+  return self;
+}
+
+- (instancetype)initWithCapacity:(NSUInteger)numItems {
+  self = [super init];
+  if (self != nil) {
+    _dictionary = [[NSMutableDictionary alloc] initWithCapacity:numItems];
+    _order = [NSMutableArray new];
+  }
+  return self;
+}
+
+- (void)setObject:(id)anObject forKey:(id<NSCopying>)aKey {
```

Comment:
If you decided to remove dictionary property based on my comment below, you can call a method of super class to take care of dictionary